esptfa_arima/Authentication/services.py

```
@@ -31,11 +38,161 @@ def register_user(username, password, first_name, last_name, email, acc_type, lr
+            middle_name = str(student.get("middle_name", "") or "").strip()
+            last_name = str(student.get("last_name")).strip()
+            section_input = student.get("section")
+
+            # Determine the section object
+            section = sections_dict.get(section_input)
+            if not section:
+                raise DRFValidationError(f"Row {index + 1}: Section '{section_input}' does not exist")
+
+            # Check for existing student
+            if Student.objects.filter(lrn=lrn, section=section).exists():
+                raise DRFValidationError(f"Row {index + 1}: Student with LRN '{lrn}' already exists in section '{section.section_name}'")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)